File: A-SOURCE_CODE/A-SERVICE/Administration/Shared/Models/Sorting.cs
using Shared.Enumerations.Order;

namespace Shared.Models
{
    public class Sorting<T>
    {
        /// <summary>
        /// Property which should be used for sorting.
        /// </summary>
        public T Property { get; set; }

        /// <summary>
        /// Whether records should be sorted ascending or decending.
        /// </summary>
        public SortDirection Direction { get; set; }
    }
}

File: A-SOURCE_CODE/B-CLIENT/A-WEB/Administration/src/models/Sorting.ts
import {SortDirection} from "../enumerations/sort-direction";

export class Sorting<T> {

  //#region Properties

  /*
  * Whether record should be sorted ascending or descending.
  * */
  public direction: SortDirection;

  /*
  * Property which should be sorted.
  * */
  public property: T;

  //#endregion
}
